src/main/java/CheeseBot.java

```diff
@@ -0,0 +1,372 @@
+            System.out.println("\tMissing /to argument");
+            return false;
+        } else if (fromIndex + 6 == toIndex) {
+            System.out.println("\tMissing /from argument");
+            return false;
+        }
+        return true;
+    }
+
+    private static boolean isValidDeadlineInput(String input) {
+        int byIndex = input.indexOf("/by ");
+        int spaceIndex = input.indexOf(" ");
+        if (byIndex == -1 || byIndex + 4 > input.length()) {
+            System.out.println("\tMissing deadline! ");
+            return false;
+        }
+        if (spaceIndex + 1 == byIndex) {
+            System.out.println("\tMissing task name!");
+            return false;
+        }
+        return true;
+    }
+
+    public static String[] parseInput(String input) {
+        String[] parsed = new String[4];
+        int spaceIndex = input.indexOf(" ");
+        if (spaceIndex == -1) {
+            parsed[0] = input;
+            return parsed;
+        }
+        String command = input.substring(0, spaceIndex);
+        String taskName;
+        parsed[0] = command;
+
+        switch (command) {
+        case "todo":
+            taskName = input.substring(spaceIndex + 1);
+            parsed[1] = taskName;
+            break;
+
+        case "deadline":
+            int byIndex = input.indexOf("/by ");
+
+            taskName = input.substring(spaceIndex + 1, byIndex - 1);
+            String by = input.substring(byIndex + 4);
+            parsed[1] = taskName;
+            parsed[2] = by;
+            break;
+
+        case "event":
+            int fromIndex = input.indexOf("/from ");
+            int toIndex = input.indexOf("/to ");
+
+            taskName = input.substring(spaceIndex + 1, fromIndex - 1);
+            String from = input.substring(fromIndex + 6, toIndex - 1);
+            String to = input.substring(toIndex + 4);
+            parsed[1] = taskName;
+            parsed[2] = from;
+            parsed[3] = to;
+            break;
+
+        case "mark":
+            //Fallthrough
+        case "unmark":
+            String taskNumber = input.substring(spaceIndex + 1);
+            parsed[1] = taskNumber;
+            break;
+        }
+        return parsed;
+    }
+
+    public static void mark(String[] arguments, boolean isDone) {
+        if (isTasksListEmpty()) {
+            return;
+        }
+
+        int taskNumber = Integer.parseInt(arguments[1]) - 1;
+        Task taskToEdit = TASKS_LIST[taskNumber];
+
+        if (isAlreadyMarked(isDone, taskToEdit) || isAlreadyUnmarked(isDone, taskToEdit)) {
+            return;
+        }
+
+        changeTaskStatus(isDone, taskToEdit, taskNumber);
+    }
+
+    private static boolean isTasksListEmpty() {
+        if (numberOfTasks == 0) {
+            System.out.println("\tThere are no tasks in your list! Please add some tasks.");
+            System.out.println("\t-------------------------------------------------------------------");
```

Comment:
Since this line separator is used quite often, would it be better to use a named constant to avoid "magic strings"?
For example,  `static final String LINE_SEPARATOR = "\t-------------------------------------------------------------------";`